src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -338,7 +338,10 @@ public PrivilegesEvaluatorResponse evaluate(
         );
 
         if (isClusterPerm(action0)) {
-            if (!securityRoles.impliesClusterPermissionPermission(action0)) {
+            // We add a check here for `impliesLegacyPermission` to ensure that no new action names miss evaluation here
+            // e.g. ad:get/detector/info will be evaluated as `cluster:admin/opensearch/ad/get/detector/info` against user's granted
+            // permissions
```

Comment:
Is that means in this case, the `get/detector/info` is the legacy permission? I was just trying to figure out where is this legacy permission coming from.

Reply:
legacy is the current nomenclature of defining actions. i.e `cluster:admin/opensearch/...`, with the new permissioning scheme we may have it defined as `ad:detector/info` but the role might still say `cluster:admin/opensearch/ad/detector/info`, and to address this mismatch I added the extra check `impliesLegacyPermissions(action0)`